src/Cabal/Plan.hs

```diff
@@ -179,6 +179,13 @@ data Unit = Unit
        -- ^ Source of the package
        --
        -- @since 0.5.0.0 (TODO)
+    , uOptional :: !(Maybe Bool)
```

Comment:
Note that this doesn't reflect the default implemented in `cabal`, with your current implementation this field would only ever be either `Nothing` or `Just True`, but never `Just False` -- this makes little sense :-)